src/State.js

```diff
@@ -217,7 +217,7 @@ class State extends EventEmitter {
 		if (config.validator) {
 			var validatorReturn = this.callFunction_(config.validator, [value, name, this]);
 
-			if (validatorReturn instanceof Error) {
+			if (process.env.NODE_ENV === 'developement' && validatorReturn instanceof Error) {
```

Comment:
Btw, I know that envify also works for webpack for example, but we have many different build processes which don't use it either, and the output wouldn't work anymore with this approach. So I prefer to have the code always work, letting people remove the calls if they wish on their builds.